### PR TITLE
Introduce python autoformatter

### PR DESCRIPTION
**Pull request by m-j-mcdonald**
This adds a python auto formatter as discussed in #472. I introduced a command 'fab format' to run the auto formatter, which also gets run as part of submit. As an example of what the auto formatter might do, look at how the code of __init__.py and git.py was modified.

### CODE REVIEW

File: fabfile/__init__.py [hunk 1]
```diff
@@ -22,7 +23,8 @@ def feature(branch_name=None):
         dependencies.check()
         migrations.check()
     else:
-        print ui.error('No branch name given. Usage: fab feature:<branch_name>')
+        print ui.error(
+            'No branch name given. Usage: fab feature:<branch_name>')
```
Comment (m-j-mcdonald) [inline fabfile/__init__.py]: This is an example of the formatting

File: fabfile/__init__.py [hunk 2]
```diff
@@ -37,8 +39,32 @@ def update():
 
 
 @task
+def format():
+    '''Format and commit python files committed on the current feature branch'''
+    diff_list = git.diff_name_only()
+    py_diff_list = [pyfile for pyfile in diff_list if pyfile[-3:] == '.py']
```
Comment (kmeht) [inline fabfile/__init__.py]: `if pyfile.endswith('.py')`

File: fabfile/__init__.py [hunk 3]
```diff
@@ -37,8 +39,32 @@ def update():
...
+            if confirm('Accept changes to %s?' % pyfile):
+                FormatFile(pyfile, in_place=True)
+    else:
+        for pyfile in py_diff_list:
+            FormatFile(pyfile, in_place=True)
+    print ui.info('Formatting complete')
+
+    if len(local('git status --porcelain', capture=True)) > 0:
+        local('git add --all')
+        local('git commit -m "Ran autoformatter"')
```
Comment (kmeht) [inline fabfile/__init__.py]: These two lines can be collapsed into `git commit -am "Ran autoformatter"`

File: fabfile/__init__.py [hunk 4]
```diff
@@ -37,8 +39,32 @@ def update():
...
+
+
+@task
 def submit(remote='origin', skip_tests=False):
     '''Push the current feature branch and create/update pull request.'''
+    execute(format)
```
Comment (kmeht) [inline fabfile/__init__.py]: What's going on here? We can't just call the function?

**Comment by kmeht** [comment]
Great diff! Does yapf seem to be the standard formatter these days? Does it do other stuff than whitespace? Do we want to include options for other formatting as well?

**Comment by m-j-mcdonald** [comment]
Autopep8 seems to still be more common but people prefer yapf. I believe it only handles whitespace, but the advantage is that instead of just fixing pep8 violations it reformats the whole file to follow a consistent style.

**Comment by kmeht** [comment]
Ok, let's see how this formatter works for us for a while. Wouldn't be too hard to swap out the formatter later if we so choose.

**Pull request merged**